viewing_party/party.py

```diff
@@ -1,23 +1,220 @@
+    friends_watched_list_set = set()
+    friends_unique_watched_list = []
+
+### if watched_list empty, return empty list
+    if not friends_watched_list:
+        return []
+### creating set of user's watched titles [same as prior]
+    for movie_dict in watched_list:
+        watched_list_set.add(movie_dict["title"])
+### creating set of user's friends' watched titles [same as prior]
+    for friend in friends_watched_list:
+        for movie_dict in friend["watched"]:
+            friends_watched_list_set.add(movie_dict["title"])
+### checking for titles in user watched_list that are not also present in friends_watched_list
+    friends_unique_watched_set = friends_watched_list_set - watched_list_set
+### if complete overlap, return empty list
+    if len(friends_unique_watched_set) == 0:
+        return []
+### else return list of dictionaries with title, genre, rating for unique items
+    for title in friends_unique_watched_set:
+        for friend in friends_watched_list:
+            for movie_dict in friend["watched"]:
+                if movie_dict["title"] == title:
+                    # for entry in friends_unique_watched_list:
+                        if movie_dict not in friends_unique_watched_list:
+                            friends_unique_watched_list.append(movie_dict)
+                    
+    return friends_unique_watched_list
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    friends_unique_watched_list = get_friends_unique_watched(user_data)
+    subscriptions_list = user_data["subscriptions"]
+    recommended_movies = []
+
+    ### iterating through movie dictionaries in friends_unique_watched_list to find movies user has 
+    ### access to via subscription
+    for movie_dict in friends_unique_watched_list:
+        # for movie_dict in friend["watched"]:
+            ### determining if a given movie that user has not seen is available via subscription service, 
+            ### if so append to recommend_movies
+        if movie_dict["host"] in subscriptions_list:
+            recommended_movies.append(movie_dict)
+
+    return recommended_movies
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    friends_unique_watched_list = get_friends_unique_watched(user_data)
+    recommended_movies = []
+    most_watched_genre = get_most_watched_genre(user_data)
+    print(most_watched_genre)
+    for movie_dict in friends_unique_watched_list:
+        ### determining if a given movie that user has not seen is of preferred genre, 
+        ### if so append to recommend_movies
+        if movie_dict["genre"] == most_watched_genre:
+            recommended_movies.append(movie_dict)
+
+    return recommended_movies
+
+
+def get_rec_from_favorites(user_data):

```

Comment:
Really well done on wave 5 y'all! I know I mentioned it before, but the re-use of your older functions is sign that you are developing the muscle of writing DRY code!